packages/compiler/src/server/server.ts

```diff
@@ -126,7 +127,18 @@ function main() {
   connection.onHover(profile(s.getHover));
   connection.onSignatureHelp(profile(s.getSignatureHelp));
   connection.onCodeAction(profile(s.getCodeActions));
-  connection.onExecuteCommand(profile(s.executeCommand));
+  // connection.onExecuteCommand(profile(s.executeCommand));
+
+  connection.onExecuteCommand(async (params) => {
+    const result = await s.executeCommand(params);
+    const [commandType, docUrl] = result;
+    if (commandType === Commands.OPEN_RULE_DOC) {
+      await connection.sendRequest("OpenLintRuleDocRequest", {
```

Comment:
the logic is more complex than i expected to support this feature. let's talk about this offline